Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,355 @@
+            {
+                comp.Injecting = false;
+                return;
+            }
+
+            var hostXform = EntityManager.GetComponent<TransformComponent>(ev.Target);
+            var host = EntityManager.EnsureComponent<GuardianHostComponent>(ev.Target);
+            // Use map position so it's not inadvertantly parented to the host + if it's in a container it spawns outside I guess.
+            var guardian = EntityManager.SpawnEntity(comp.GuardianProto, hostXform.MapPosition);
+
+            host.GuardianContainer.Insert(guardian);
+            host.HostedGuardian = guardian;
+
+            if (TryComp(guardian, out GuardianComponent? guardianComponent))
+            {
+                guardianComponent.Host = ev.Target;
+
+                // Grant the user the recall action and notify them
+                actions.Grant(ActionType.ManifestGuardian);
+                SoundSystem.Play(Filter.Entities(ev.Target), "/Audio/Effects/guardian_inject.ogg", ev.Target);
+
+                _popupSystem.PopupEntity(Loc.GetString("guardian-created"), ev.Target, Filter.Entities(ev.Target));
+                // Exhaust the activator
+                comp.Used = true;
+            }
+            else
+            {
+                Logger.ErrorS("guardian", $"Tried to spawn a guardian that doesn't have {nameof(GuardianComponent)}");
+                EntityManager.QueueDeleteEntity(guardian);
+            }
+        }
+
+        /// <summary>
+        /// Triggers when the host receives damage which puts the host in either critical or killed state
+        /// </summary>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (!component.HostedGuardian.IsValid()) return;
+
+            if (args.CurrentMobState.IsCritical())
+            {
+                _popupSystem.PopupEntity(Loc.GetString("guardian-critical-warn"), component.HostedGuardian, Filter.Entities(component.HostedGuardian));
+                SoundSystem.Play(Filter.Entities(component.HostedGuardian), "/Audio/Effects/guardian_warn.ogg", component.HostedGuardian);
+            }
+            else if (args.CurrentMobState.IsDead())
+            {
+                SoundSystem.Play(Filter.Pvs(uid), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.20f));
+                EntityManager.RemoveComponent<GuardianHostComponent>(uid);
+            }
+        }
+
+        /// <summary>
+        /// Handles guardian receiving damage and splitting it with the host according to his defence percent
+        /// </summary>
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            if (args.DamageDelta == null ||
+                !HasComp<DamageableComponent>(uid) ||
+                !TryComp<DamageableComponent>(component.Host, out var hostDamage)) return;
```

Comment:
Given that this event was directed at `uid`, then `HasComp<DamageableComponent>(uid)` should always be true.

Reply:
It's not raised directed for that component though?

Reply:
At least currently, the event is only ever raised by `DamageableSystem` directed at an entity that has a `DamageableComponent`. The component is even passed to the `DamageChangedEvent` constructor. So `HasComp<DamageableComponent>(uid)` should always be true.

But even if at some point that changes (e.g., a body system, where an entity is not directly damageable but contains damageable limbs), I don't see why the `HasComponent` check is needed.